File: genefunk/src/main/java/de/hipp/pnp/genefunk/GeneFunkClassService.java
package de.hipp.pnp.genefunk;

import org.springframework.stereotype.Service;

import javax.annotation.PostConstruct;
import java.util.List;

@Service
public class GeneFunkClassService {

    final GeneFunkClassRepository repository;
    final GeneFunkClassFactory characterClassFactory;

    public GeneFunkClassService(GeneFunkClassRepository repository, GeneFunkClassFactory characterClassFactory) {
        this.repository = repository;
        this.characterClassFactory = characterClassFactory;
    }


    public List<GeneFunkClass> getAllClasses() {
        return repository.findAll();
    }

    @PostConstruct
    private void populateData() {
        repository.saveAll(characterClassFactory.initiateClasses());
    }
}

File: genefunk/src/main/kotlin/de/hipp/pnp/genefunk/GeneFunkClassService.kt
package de.hipp.pnp.genefunk

import de.hipp.pnp.base.entity.GeneFunkClass
import de.hipp.pnp.base.rabbitmq.GenefunkInfoProducer
import io.github.oshai.kotlinlogging.KotlinLogging
import jakarta.annotation.PostConstruct
import org.springframework.stereotype.Service

@Service
class GeneFunkClassService(
    private val genefunkInfoProducer: GenefunkInfoProducer,
) {
    var genefuncClasses = mutableMapOf<String, GeneFunkClass>()

    private val log = KotlinLogging.logger {}

    @PostConstruct
    fun init() {
        val classes: Map<String, GeneFunkClass> = genefunkInfoProducer.getAllClasses()
        if (classes.isNullOrEmpty()) {
            log.warn { "No classes found for GeneFunk" }
            return
        }

        this.genefuncClasses.putAll(classes)
        log.info { "Initialized GeneFunkClassService with ${classes.size} classes" }
    }

    fun getAllClasses(): MutableMap<String, GeneFunkClass> = genefuncClasses
}
